packages/luigis-box/src/Model/Batch/LuigisBoxBatchLoader.php

```diff
@@ -0,0 +1,105 @@
+use Shopsys\LuigisBoxBundle\Component\LuigisBox\LuigisBoxResult;
+
+class LuigisBoxBatchLoader
+{
+    /**
+     * @var array<string, int>
+     */
+    protected static array $totalsByType = [];
+
+    /**
+     * @param \Shopsys\LuigisBoxBundle\Component\LuigisBox\LuigisBoxClient $luigisBoxClient
+     * @param \GraphQL\Executor\Promise\PromiseAdapter $promiseAdapter
+     * @param \Shopsys\FrameworkBundle\Model\Product\Search\ProductElasticsearchRepository $productElasticsearchRepository
+     * @param \Shopsys\FrameworkBundle\Model\Product\Search\FilterQueryFactory $filterQueryFactory
+     */
+    public function __construct(
+        protected readonly LuigisBoxClient $luigisBoxClient,
+        protected readonly PromiseAdapter $promiseAdapter,
+        protected readonly ProductElasticsearchRepository $productElasticsearchRepository,
+        protected readonly FilterQueryFactory $filterQueryFactory,
+    ) {
+    }
+
+    /**
+     * @param string $type
+     * @return int
+     */
+    public static function getTotalByType(string $type): int
+    {
+        return self::$totalsByType[$type] ?? 0;
+    }
+
+    /**
+     * @param \Shopsys\LuigisBoxBundle\Model\Batch\LuigisBoxBatchLoadData[] $luigisBoxBatchLoadData
+     * @return \GraphQL\Executor\Promise\Promise
+     */
+    public function loadByBatchData(array $luigisBoxBatchLoadData): Promise
+    {
+        $query = '';
+        $limitsByType = [];
+        $endpoint = '';
+        $page = 0;
+        $filter = [];
+        $orderingMode = null;
+
+        foreach ($luigisBoxBatchLoadData as $luigisBoxBatchLoadDataItem) {
+            if ($luigisBoxBatchLoadDataItem->getType() === LuigisBoxClient::MAIN_TYPE) {
+                $query = $luigisBoxBatchLoadDataItem->getQuery();
+                $endpoint = $luigisBoxBatchLoadDataItem->getEndpoint();
+                $page = $luigisBoxBatchLoadDataItem->getPage();
+                $filter = $luigisBoxBatchLoadDataItem->getFilter();
+                $orderingMode = $luigisBoxBatchLoadDataItem->getOrderingMode();
+            }
+
+            $limitsByType[$luigisBoxBatchLoadDataItem->getType()] = $luigisBoxBatchLoadDataItem->getLimit();
+        }
+
+        return $this->promiseAdapter->all($this->mapDataByTypes(
+            $this->luigisBoxClient->getData($query, $endpoint, $page, $limitsByType, $filter, $orderingMode),
+            $limitsByType,
+        ));
+    }
+
+    /**
+     * @param \Shopsys\LuigisBoxBundle\Component\LuigisBox\LuigisBoxResult[] $luigisBoxResults
+     * @param array $limitsByType
+     * @return array
+     */
+    protected function mapDataByTypes(array $luigisBoxResults, array $limitsByType): array
+    {
+        $mappedData = [];
+
+        foreach ($limitsByType as $type => $limit) {
+            $methodNameForDataMapping = 'map' . ucfirst($type) . 'Data';
+            $mappedDataOfCurrentType = $this->{$methodNameForDataMapping}($luigisBoxResults[$type], $limit);
```

Comment:
I'm not sure about this approach... 
I don't like that the method usage is not clear for static analysis and ide... 

I would rather have the explicit mapping method

Reply:
I have changed it to if